algorithms/CPlusPlus/Arrays/search-sorted-rotated.cpp

```diff
@@ -0,0 +1,48 @@
+//Description : Searching element in a sorted-rotated array using binary search
+#include<bits/stdc++.h>
+using namespace std;
+int search_array(vector <int> arr, int item) {
+    int low = 0;
+    int high = arr.size() - 1;
+    while (low <= high) {
+        int mid = (low + high) / 2;                             //Calculating middle index
+        if (arr[mid] == item) {                                 //If the element at middle index is equal to the item then we return the middle index
```

Comment:
for finding mid-value replace line 8 with:
`int mid = low + (high-low)/2;`
Because the logic that you wrote might fail test cases for a larger number.